### PR TITLE
Adding Hyperspectral Image Classification

### PR DESCRIPTION
**Pull request by aish160**
### **Purpose**
Hyperspectral Image Classification using SpinalNet model and Remote sensing Dataset

### **Changes**

- Adding a folder which consists of IPYNB Notebook, Python file, Classification Map, Confusion Matrix, Ground Truth.

- Experimented SpinalNet model with Kennedy Space Center Dataset and also used PCA for dimentionality reduction.

### **Context**
With the advancement of remote sensing technology, the utilization of Hyperspectral images is getting increasingly broad. The exact arrangement of ground includes through Hyperspectral pictures which is a significant research content and has attracted in far and wide consideration. The objective of Hyperspectral imaging is to acquire the range for every pixel in the picture of a scene, to discover objects that have 100 to 200 sorts of these groups. It gives a greater Color information for every pixel by recognizing materials, or detecting processes.

The SpinalNet model was experimented on a Remote Sensing Dataset i.e, [Kennedy Space Center](http://www.ehu.eus/ccwintco/index.php/Hyperspectral_Remote_Sensing_Scenes#Kennedy_Space_Center_.28KSC.29) (KSC). The KSC data, acquired from an altitude of approximately 20 km, have a spatial resolution of 18 m. After removing water absorption and low SNR bands, 176 bands were used for the analysis. Training data were selected using land cover maps derived from color infrared photography provided by the Kennedy Space Center and Landsat Thematic Mapper (TM) imagery. Discrimination of land cover for this environment is difficult due to the similarity of spectral signatures for certain vegetation types. For classification purposes, 13 classes representing the various land cover types that occur in this environment were defined for the site.

### **SpinalNet Model**
SpinalNet is a recently evolving model. The main Objective of this Spinal Net model is, it learns high-level abstract features by using the CNN architecture. Comparable to our spinal cord, the preffered  SpinalNet takes inputs progressively and redundantly. Every layer of the SpinalNet supply into the local output i.e, reflex. The SpinalNet additionally dispatches a tweaked variant of contributions supply into the global output i.e, Brain. The NN training measure arranges loads depending on the training data, like our Brain, designing the spinal neurons for arranging the torment affectability of various nerve system of our body.
As mentioned above about the features of human spinal cord, SpinalNet has the following similarities:

- It has Gradual input

- It has local output and also probable global influence

- and the weights are reconfigured during training

Generally in a neural network ,the first hidden layer receives the input and then transfers the outcomes to the remaining hidden layers and then to the output layer. But in SpinalNet, the output layer adds the weights of all hidden layers. The SpinalNet consists of Input layer, Intermediate hidden layers and Output layers. The input is split and sent to the multiple hidden layers. Every layer receives the previous layer’s output except the first layer. The SpinalNet takes inputs gradually and repetitively. The weights of the neural network are configured based on the training data similar to the brain configuring for pain to the spinal nerves. The number of output neurons is equal to the number of output neurons per each hidden layer. If one feature doesn't have any impact on one of the intermediate layers, it may have an impact on any other intermediate layers.

### **Results**
SpinalNet model with KSC dataset accomplished promising accuracy of 95.40%. Hence, SpinalNet performance on remote sensing datasets is outstanding.

Download MATLAB data file: [Kennedy Space Center (KSC)](http://www.ehu.es/ccwintco/uploads/2/26/KSC.mat) (56.8 MB)
Download MATLAB ground truth file: [KSC gt](http://www.ehu.es/ccwintco/uploads/a/a6/KSC_gt.mat) (3.2 kB)

### CODE REVIEW

**Review by Kaushikrudra — APPROVED** [review]

**Comment by 1AyaNabil1** [comment]
you are great
